### PR TITLE
Change test storage from List to class in Fix plugin

### LINKED ISSUE #241: Change test storage from List to class in Fix plugin
Create an interface and a field with the type of this interface that would contain TEST field.
Plugins would need to implement and extend this class with other fields (like Expected) for Fix plugin

### PR DESCRIPTION
**Pull request by Cheshiriks**
What's done:
* Change test storage from List to class in plugin
Closes #241

### CODE REVIEW

File: save-plugins/fix-plugin/src/commonMain/kotlin/org/cqfn/save/plugins/fix/FixPlugin.kt [hunk 1]
```diff
@@ -171,4 +171,20 @@ class FixPlugin(
         }
         .toList()
         .joinToString { (type, lines) -> "$type: $lines lines" }
+
+    /**
+     * @property expected expected file
+     */
+    interface FixTestFiles : TestFiles {
```
Comment (akuleshov7) [inline save-plugins/fix-plugin/src/commonMain/kotlin/org/cqfn/save/plugins/fix/FixPlugin.kt:178]: Why is this needed?

File: save-plugins/fix-plugin/src/commonMain/kotlin/org/cqfn/save/plugins/fix/FixPlugin.kt [hunk 2]
```diff
@@ -171,4 +171,20 @@ class FixPlugin(
...
+         * path to expected file
+         */
+        val expected: Path
+    }
+
+    /**
+     * @property test test file
+     * @property expected expected file
+     */
+    class Test(override val test: Path, override val expected: Path) : FixTestFiles
```
Comment (akuleshov7) [inline save-plugins/fix-plugin/src/commonMain/kotlin/org/cqfn/save/plugins/fix/FixPlugin.kt:189]: why not a data class?

**Review by akuleshov7 — CHANGES_REQUESTED** [review]
please remove dummy structures

File: save-plugins/fix-plugin/src/commonMain/kotlin/org/cqfn/save/plugins/fix/FixPlugin.kt (hunk 1 above)
Comment (Cheshiriks) [reply save-plugins/fix-plugin/src/commonMain/kotlin/org/cqfn/save/plugins/fix/FixPlugin.kt]: Fixed

File: save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt [hunk 3]
```diff
@@ -85,25 +85,22 @@ abstract class Plugin(
             }
 
         return if (testFiles.isNotEmpty()) {
-            rawTestFiles.filter { resourcesGroup ->
-                // test can be specified by the name of one of it's files
-                resourcesGroup.any { path ->
-                    testFiles.any { it in path.toString() }
-                }
+            rawTestFiles.filter { path ->
```
Comment (petertrr) [inline save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt:88]: Pls rename `path` argument

File: save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt [hunk 4]
```diff
@@ -85,25 +85,22 @@ abstract class Plugin(
...
             rawTestFiles
         }
     }
 
-    private fun isExcludedTest(testFiles: List<Path>, excludedTests: List<String>?): Boolean {
+    private fun isExcludedTest(testFiles: TestFiles, excludedTests: List<String>?): Boolean {
         // common root of the test repository (not a location of a current test)
         val testRepositoryRoot = testConfig.getRootConfig().location
         // creating relative to root path from a test file
         // FixMe: https://github.com/cqfn/save/issues/241 here we are incorrectly using testFiles[0], as for example it is
```
Comment (petertrr) [inline save-common/src/commonMain/kotlin/org/cqfn/save/core/plugin/Plugin.kt:100]: Pls remove this comment

File: save-plugins/fix-plugin/src/commonMain/kotlin/org/cqfn/save/plugins/fix/FixPlugin.kt [hunk 5]
```diff
@@ -46,12 +46,12 @@ class FixPlugin(
...
     @Suppress("TOO_LONG_FUNCTION")
-    override fun handleFiles(files: Sequence<List<Path>>): Sequence<TestResult> {
+    override fun handleFiles(files: Sequence<TestFiles>): Sequence<TestResult> {
         val fixPluginConfig = testConfig.pluginConfigs.filterIsInstance<FixPluginConfig>().single()
         val generalConfig = testConfig.pluginConfigs.filterIsInstance<GeneralConfig>().single()
 
-        return files.chunked(fixPluginConfig.batchSize!!.toInt()).map { chunk ->
-            val pathMap = chunk.map { it.first() to it.last() }
+        return files.map { it as Test }.chunked(fixPluginConfig.batchSize!!.toInt()).map { chunk ->
+            val pathMap = chunk.map { it.expected to it.test }
```
Comment (petertrr) [inline save-plugins/fix-plugin/src/commonMain/kotlin/org/cqfn/save/plugins/fix/FixPlugin.kt:54]: Shouldn't it be vice versa? It was the original problem leading to this change

File: save-plugins/fix-plugin/src/commonMain/kotlin/org/cqfn/save/plugins/fix/FixPlugin.kt [hunk 6]
```diff
@@ -171,4 +171,10 @@ class FixPlugin(
         }
         .toList()
         .joinToString { (type, lines) -> "$type: $lines lines" }
+
+    /**
+     * @property test test file
+     * @property expected expected file
+     */
+    class Test(override val test: Path, val expected: Path) : TestFiles
```
Comment (petertrr) [inline save-plugins/fix-plugin/src/commonMain/kotlin/org/cqfn/save/plugins/fix/FixPlugin.kt:179]: Different name? It looks confusing in `FixAndWarnPlugin`

**Review by petertrr — CHANGES_REQUESTED** [review]

File: save-plugins/fix-plugin/src/commonMain/kotlin/org/cqfn/save/plugins/fix/FixPlugin.kt (hunk 5 above)
Comment (Cheshiriks) [reply save-plugins/fix-plugin/src/commonMain/kotlin/org/cqfn/save/plugins/fix/FixPlugin.kt]: Fixed

**Review by akuleshov7 — APPROVED** [review]

**Pull request merged**